### PR TITLE
Build: Remove xpack specific run task

### PR DESCRIPTION
**Pull request by rjernst**
With the opening of xpack, we still retained a run task within
:x-pack:plugin. However, the root level run task also runs with the
default distribution. This change removes the extra run task inside
xpack in favor of using the root level task, and moves the
license/configuration code for run into the main run configuration.

### CODE REVIEW

File: x-pack/plugin/build.gradle [hunk 1]
```diff
@@ -8,16 +8,8 @@ import java.nio.file.Path
 import java.nio.file.StandardCopyOption
 import org.elasticsearch.gradle.test.RunTask;
 
-apply plugin: 'elasticsearch.es-meta-plugin'
```
Comment (nik9000) [inline x-pack/plugin/build.gradle]: I talked with @rjernst about this part - now that xpack is bundled as part of the default distribution the run task was actually the last part that needed this.

**Review by nik9000 — APPROVED** [review]

**Comment by nik9000** [comment]
I hadn't noticed that @rjernst hadn't yet put `review` on this. I like it in theory, once he's slayed the gradle demons.

File: x-pack/plugin/build.gradle (hunk 1 above)
Comment (rjernst) [reply x-pack/plugin/build.gradle]: Sorry for the confusion. I only meant to move the run task configuration in this PR. This removal was accidental, and from another branch.

**Review by nik9000 — APPROVED** [review]
That is more like it!

**Review by hub-cap — APPROVED** [review]
:shipit:

**Pull request merged**